generators/entity-server/templates/dotnetcore/src/Project/Controllers/EntityController.cs.ejs

```diff
@@ -128,6 +132,9 @@ namespace <%= namespace %>.Controllers {
             <%_ let first = true;
             let suffix = 0;
             relationships.forEach( relationship => {
+                if (relationship.otherEntityNamePascalized === 'User' && relationship.relationshipType !== 'many-to-many') { _%>
+            <%= camelCasedEntityClass %>.User = null;
```

Comment:
Maybe check if the user saved at database is the same user of the request?
Throw security-related exception if user is different. 
Maybe check if user is admin (maybe admin can chage user relationship)?

Reply:
It's just a relation to the user, i think all user can set this ( It's the same in java version) . @manisi Can you confirm ?

Reply:
> It's just a relation to the user, i think all user can set this ( It's the same in java version) . @manisi Can you confirm ?

Hello!
same error exist:       : An unhandled exception has occurred while executing the request.
System.InvalidOperationException: Unable to track an entity of type 'UserRole' because primary key property 'RoleId' is null.
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.NullableKeyIdentityMap`1.Add(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.StartTracking(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState oldState, EntityState newState, Boolean acceptChanges, Boolean modifyProperties)

Reply:
@nicolas63  Forget my last comment, no need to check is admin, etc...

What is happening is that the dto/automapper implementation tries to map your dto user back to user entity (including roles), which is not desired, When the user has a role, entity framework complains that the roles entity does not have id (UserDto only has a list<string> of roles so there is no way for automapper to know the role id).

The main generator uses userId/userName on the relationship not the full User Dto:
`
{
  "id" : 14,
  "name" : "sample test",
  "userId" : 4,
  "userLogin" : null
}
`

Maybe change dto to use UserId instead of full UserDto.

Maybe load the user entity using the UserId on the controller and/or service like:
`
entity.User =  await _userManager.FindByIdAsync(entityDto.UserId);
`